### PR TITLE
fix(logo): Use the new Firefox logo in Fx Desktop/Android >= 57, iOS >= 10.

### LINKED ISSUE #5588: Update to new Firefox logo?
IIUC there's a snazzy new Firefox logo coming with the 57 release.  @ryanfeeley should we update the versions of the logo used in this repo to match?  I couldn't find an existing ticket for it.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Do some UA sniffing on startup to add the "fx-57" class
to the HTML element. If "fx-57" exists, display the
new logo.

fixes #5588

@ryanfeeley - favicon and logo match.

Fx Desktop or Android <= 56 see old logo, favicon.
Fx Desktop or Android >= 57 see new logo, favicon.

screen shot 2017-10-20 at 14 25 31
screen shot 2017-10-20 at 14 26 34

Fx iOS <= 9 see old logo, favicon
Fx iOS >= 10 see new logo, favicon

screen shot 2017-10-20 at 14 39 14

I'm not sure how to use the simulator to view Fx for iOS 9.

@mozilla/fxa-devs - r?
@ryanfeeley - r?

### CODE REVIEW

File: tests/server/routes/get-favicon.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                'user-agent': uaStrings.desktop_firefox_57
+              },
+              url: 'favicon.ico'
+            };
+            return new Promise((resolve) => {
+              instance.process(request, response, resolve);
+            });
+          },
+
+          'should see old icon': function () {
```
Comment (vbudhram) [inline tests/server/routes/get-favicon.js]: nit: `new icon`?

File: tests/server/routes/get-favicon.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                'user-agent': uaStrings.android_firefox_57
+              },
+              url: 'favicon.ico'
+            };
+            return new Promise((resolve) => {
+              instance.process(request, response, resolve);
+            });
+          },
+
+          'should see old icon': function () {
```
Comment (vbudhram) [inline tests/server/routes/get-favicon.js]: Same as above, new icon?

File: tests/server/routes/get-favicon.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                'user-agent': uaStrings.ios_firefox_10
+              },
+              url: 'favicon.ico'
+            };
+            return new Promise((resolve) => {
+              instance.process(request, response, resolve);
+            });
+          },
+
+          'should see old icon': function () {
```
Comment (vbudhram) [inline tests/server/routes/get-favicon.js]: nit: `new icon`?

File: tests/server/routes/get-favicon.js (hunk 1 above)
Comment (shane-tomlinson) [reply tests/server/routes/get-favicon.js]: You are right!

**Pull request merged**